src/net.zig

```diff
@@ -0,0 +1,746 @@
+    ///
+    /// ## Thread Safety
+    /// It is safe to call this function from any thread.
+    ///
+    /// ## Version
+    /// This function is available since SDL_net 3.0.0.
+    pub fn getStatus(self: Address) Error!Status {
+        const ret = c.NET_GetAddressStatus(self.value);
+        if (ret < 0) {
+            errors.callErrorCallback();
+            return error.SdlNetError;
+        }
+        return if (ret == 1) .resolved else .resolving;
+    }
+
+    /// Get a human-readable string from a resolved address.
+    ///
+    /// ## Return Value
+    /// Returns a string, or `null` on error or if not yet resolved.
+    ///
+    /// ## Remarks
+    /// The returned string is owned by the `net.Address` and is valid as long as the object lives.
+    ///
+    /// ## Thread Safety
+    /// It is safe to call this function from any thread.
+    ///
+    /// ## Version
+    /// This function is available since SDL_net 3.0.0.
+    pub fn getString(self: Address) ?[:0]const u8 {
+        const ret = c.NET_GetAddressString(self.value);
+        if (ret == null) return null;
+        return std.mem.sliceTo(ret, 0);
+    }
+
+    /// Compare two `net.Address` objects.
+    ///
+    /// ## Function Parameters
+    /// * `self`: first address to compare.
+    /// * `other`: second address to compare.
+    ///
+    /// ## Return Value
+    /// Returns `std.math.Order.lt` if `self` is "less than" `other`, `std.math.Order.gt` if "greater than", `std.math.Order.eq` if equal.
+    ///
+    /// ## Thread Safety
+    /// It is safe to call this function from any thread.
+    ///
+    /// ## Version
+    /// This function is available since SDL_net 3.0.0.
+    pub fn compare(self: Address, other: Address) std.math.Order {
+        return std.math.order(c.NET_CompareAddresses(self.value, other.value), 0);
+    }
+};
+
+/// Enable simulated address resolution failures.
+///
+/// ## Function Parameters
+/// * `percent_loss`: A number between 0 and 100. Higher means more failures. Zero to disable.
+///
+/// ## Thread Safety
+/// It is safe to call this function from any thread.
+///
+/// ## Version
+/// This function is available since SDL_net 3.0.0.
+pub fn simulateAddressResolutionLoss(percent_loss: u8) void {
+    c.NET_SimulateAddressResolutionLoss(@intCast(percent_loss));
+}
+
+/// Obtain a list of local addresses on the system.
+///
+/// ## Function Parameters
+/// * `allocator`: The allocator to use for the returned slice.
+///
+/// ## Return Value
+/// Returns a slice of `net.Address` pointers, one for each bindable address on the system.
+/// The caller owns the slice and must free it. The caller also must call `deinit()` on each `net.Address` in the slice.
+///
+/// ## Thread Safety
+/// It is safe to call this function from any thread.
+///
+/// ## Version
+/// This function is available since SDL_net 3.0.0.
+pub fn getLocalAddresses(allocator: std.mem.Allocator) Error![]Address {
```

Comment:
Is there a way we can avoid this? Since it's an opaque handle, we can just make the `Address` be packed and assert its size being the same as an SDL address as `comptime`. I have some subsystems that do this.